### ISSUE TITLE
Avoid unnecessary includes

### ISSUE THREAD

**Issue by ghseb** [PelionIoT/mbed-cloud-client#93]
Hello,

when `MBED_CLOUD_CLIENT_SUPPORT_UPDATE` is not defined, `UpdateClientResources.h` and `update_client_public.h` should probably not be included in `ServiceClient.cpp` to avoid unnecessary dependencies towards update client hub.

https://github.com/PelionIoT/mbed-cloud-client/blob/d7edc529ed3722c811ff401440ef58ea980bf543/source/ServiceClient.cpp#L28-L31

Proposal:
```
#if !defined(MBED_CLOUD_CLIENT_FOTA_ENABLE)
#if defined(MBED_CLOUD_CLIENT_SUPPORT_UPDATE)
#include "include/UpdateClientResources.h"
#include "update-client-hub/update_client_public.h"
#endif
#else
```

**Comment by teetak01** [comment]
Thanks for the feedback @ghseb. Looks sensible.

**Comment by teetak01** [comment]
https://armiot.atlassian.net/browse/IOTCLT-5026

**Comment by teetak01** [comment]
Should be fixed now with 4.11.0 release.

**Comment by ghseb** [comment]
Thanks!
